<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Arguments -->
  <arg name="initial_ts_state_from_agent" default="False"/>
  <arg name="agent_name" default="turtlebot"/>
  
  <param name="agent_name" type="string" value="$(arg agent_name)" />
  <!-- hard and soft task parameters -->
  <rosparam command="load" file="$(find ltl_automaton_planner)/config/test_ltl_formula.yaml" />
  <param name="initial_beta" value="1000" />
  <!-- Transition system -->
  <param name="transition_system_textfile" type="str" textfile="$(find ltl_automaton_planner)/config/test_ts.yaml" />

  <!-- Run planner node -->
  <node name="ltl_planner" pkg="ltl_automaton_planner" type="planner_node.py" output="screen">
    <param name="initial_ts_state_from_agent" type="bool" value="$(arg initial_ts_state_from_agent)" />
  </node>

  <!-- Run dynamic parameter node -->
  <node name="dynamic_parameter" pkg="ltl_automaton_planner" type="dynamic_parameter_node.py" output="screen">
  </node>
  
</launch>
